### PR TITLE
Add DependencyConfiguration for Distribute http client

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m [hunk 1]
```diff
@@ -2632,4 +2633,24 @@ - (void)testOpenURLInAuthenticationSessionFails API_AVAILABLE(ios(11)) {
   [(id)authenticationSessionMock stopMocking];
 }
 
+- (void)testDependencyCallUsesInjectedHttpClient {
+    
+    id httpClientMock = [MSHttpClient new];
+    [MSDependencyConfiguration setHttpClient:httpClientMock];
+    
+    // When
```
Comment (michaelsmitha) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m]:
```suggestion
    // If
```
right?

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m [hunk 2]
```diff
@@ -2632,4 +2633,24 @@ - (void)testOpenURLInAuthenticationSessionFails API_AVAILABLE(ios(11)) {
...
+    id distributeMock = OCMPartialMock(self.sut);
```
Comment (MatkovIvan) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m]: Call stop mock explicitly

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m [hunk 3]
```diff
@@ -2632,4 +2633,24 @@ - (void)testOpenURLInAuthenticationSessionFails API_AVAILABLE(ios(11)) {
...
+    self.sut.appSecret = kMSTestAppSecret;
+    id reachabilityMock = OCMClassMock([MS_Reachability class]);
+    OCMStub([reachabilityMock reachabilityForInternetConnection]).andReturn(reachabilityMock);
+    OCMStub([reachabilityMock currentReachabilityStatus]).andReturn(ReachableViaWiFi);
+   
+    // When
+    [self.sut checkLatestRelease:kMSTestUpdateToken distributionGroupId:kMSTestDistributionGroupId releaseHash:kMSTestReleaseHash];
+    
+    // Then
+    OCMReject([MSHttpClient new]);
```
Comment (jaeklim) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m]:
```suggestion
    OCMReject([MSHttpClient new]);
    [reachabilityMock stopMocking];
    [distributeMock stopMocking];
```

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m [hunk 4]
```diff
@@ -2632,4 +2633,24 @@ - (void)testOpenURLInAuthenticationSessionFails API_AVAILABLE(ios(11)) {
...
+- (void)testDependencyCallUsesInjectedHttpClient {
+    
+    id httpClientMock = [MSHttpClient new];
+    [MSDependencyConfiguration setHttpClient:httpClientMock];
+    
+    // If
+    id distributeMock = OCMPartialMock(self.sut);
+    OCMReject([distributeMock handleUpdate:OCMOCK_ANY]);
+    self.sut.appSecret = kMSTestAppSecret;
+    id reachabilityMock = OCMClassMock([MS_Reachability class]);
```
Comment (jaeklim) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m]: You must call `stopMocking` for class mocks.

**Comment by jaeklim** [comment]
@annakocheshkova Your unit test crashed once. Please check if you are adding a flaky test.

**Pull request merged**